tkg/client/management_components.go

```diff
@@ -47,8 +48,58 @@ func (c *TkgClient) InstallOrUpgradeKappController(kubeconfig, kubecontext strin
 	return err
 }
 
+// RemoveObsoleteManagementComponents lists and removes management cluster components that are obsoleted by new
+// management packages, e.g. the tkr-controller-manager deployment in tkr-system namespace.
+func RemoveObsoleteManagementComponents(clusterClient clusterclient.Client) error {
+	objectsToDelete, err := listObjectsToDelete(clusterClient)
+	if err != nil {
+		return err
+	}
+
+	for _, object := range objectsToDelete {
+		err := clusterClient.DeleteResource(object)
+		if kerrors.FilterOut(err, apierrors.IsNotFound) != nil {
+			return errors.Wrapf(err, "unable to delete resource %s: '%s/%s'",
+				object.GetObjectKind().GroupVersionKind(), object.GetNamespace(), object.GetName())
+		}
+	}
+	return nil
+}
+
+func listObjectsToDelete(clusterClient clusterclient.Client) ([]client.Object, error) {
+	var objectsToDelete []client.Object
+
+	kindsOfObjectsToDelete := map[schema.GroupVersionKind][]client.ListOption{
+		{
+			Group:   "addons.cluster.x-k8s.io",
+			Version: "v1beta1",
+			Kind:    "ClusterResourceSet",
+		}: {client.InNamespace(constants.TkrNamespace)},
+		{
+			Group:   "apps",
+			Version: "v1",
+			Kind:    "Deployment",
+		}: {client.InNamespace(constants.TkrNamespace)},
+	}
+
```

Comment:
Are these all we needed to cleanup? During my manual tests it seems that the new tkr won't be synced unless
* Delete tkr-<tkr version> configmap
* Delete all old tkrs

Reply:
Hey Chenrui, there's no need to delete old TKRs. They should probably stay, as they might have been used to create (legacy-style) clusters.

Any new TKRs will be created from TKR packages fetched by the new TKR Source Controller.

Also, TKR BOM ConfigMaps will be saved in `tog-system` namespace now, as well as replicated to `tkr-system` for the sake of backwards compatibility. See #3753